### PR TITLE
chore(gitlint): conventional commit

### PR DESCRIPTION
**Pull request by fouge**
push for conventional commits

[default types](https://jorisroovers.com/gitlint/latest/rules/contrib_rules/#ct1-contrib-title-conventional-commits): `fix,feat,chore,docs,style,refactor,perf,test,revert,ci,build`

### CODE REVIEW

File: utils/format/.gitlint [hunk 1]
```diff
@@ -11,6 +11,9 @@ ignore-squash-commits=false
 # Enable debug mode (prints more output). Disabled by default
 debug = false
 
+# enable conventional commits
+contrib = contrib-title-conventional-commits,CC1
```
Comment (Qbicz) [inline utils/format/.gitlint:15]: I think we actually want CT1, not CC1

**Review by Qbicz — CHANGES_REQUESTED** [review]

File: utils/format/.gitlint (hunk 1 above)
Comment (Qbicz) [reply utils/format/.gitlint:15]: https://jorisroovers.com/gitlint/latest/rules/contrib_rules/#ct1-contrib-title-conventional-commits

**Comment by Qbicz** [comment]
> My 2 cents: I think we should use a solution that operates at the PR level, not at the commit level. My concern with doing it per-commit is that precommit hooks are more of a hassle to set up (we don't use them in other repos, although we do use them here) and its a bit annoying to need to properly format all commits. Especially when they just get squashed at the end anyway.

I'm ok with applying it for every commit, if we can quickly mark commits as fixups. `fixup!` at the beginning should let you pass the test (such commit is created if you use `--fixup`), and then remove those during squashing.

**Comment by fouge** [comment]
ideally, one PR = one commit with the squash rules, so i think it makes sense, but i also agree that we should check the PR title

File: utils/format/.gitlint (hunk 1 above)
Comment (fouge) [reply utils/format/.gitlint]: ty

**Pull request merged**